apiserver/facades/client/action/interface.go

```diff
@@ -16,9 +16,17 @@ type State interface {
 	AllMachines() ([]*state.Machine, error)
 	Application(name string) (*state.Application, error)
 	FindEntity(tag names.Tag) (state.Entity, error)
-	Model() (Model, error)
 	WatchActionLogs(actionId string) state.StringsWatcher
 	ActionByTag(tag names.ActionTag) (state.Action, error)
+	AddAction(receiver state.ActionReceiver, operationID, name string, payload map[string]interface{}, parallel *bool, executionGroup *string) (state.Action, error)
+	EnqueueOperation(summary string, count int) (string, error)
+	FailOperationEnqueuing(operationID, failMessage string, count int) error
+	FindActionsByName(name string) ([]state.Action, error)
+	ListOperations(actionNames []string, actionReceivers []names.Tag, operationStatus []state.ActionStatus,
+		offset, limit int,
+	) ([]state.OperationInfo, bool, error)
+	OperationWithActions(id string) (*state.OperationInfo, error)
+	Type() state.ModelType
```

Comment:
As part of this PR we  also need to go through and find the uses of this `Type` func and change them. We can do this by giving the facade the model info service which you are able to get this information off of.

See https://github.com/juju/juju/blob/a6bf746419e9afa69d0ab470b1d01f43469ab48e/apiserver/facades/client/application/deploy.go#L117 for example of how we have done this previously.